src/Microsoft.ML.Transforms/Text/TextTransform.cs

```diff
@@ -116,6 +124,92 @@ public sealed class Arguments : TransformInputBase
             public TextNormKind VectorNormalizer = TextNormKind.L2;
         }
 
+        private readonly string[] _inputColumns;
+        public IReadOnlyCollection<string> InputColumns => _inputColumns.AsReadOnly();
+        public readonly string OutputColumn;
+        public readonly Language TextLanguage;
+        public readonly CaseNormalizationMode TextCase;
+        public readonly bool KeepDiacritics;
+        public readonly bool KeepPunctuations;
+        public readonly bool KeepNumbers;
+        public readonly bool OutputTokens;
+        public readonly TextNormKind VectorNormalizer;
+
+        // Mutable parameters.
+        private IStopWordsRemoverFactory _stopWordsRemover;
+        public readonly StopwordsRemoverSettings StopWordsRemover;
+        private TermLoaderArguments _dictionary;
+        public readonly TermDictionarySettings TermDictionary;
+        private INgramExtractorFactoryFactory _wordFeatureExtractor;
+        public readonly NgramExtractorSettings WordFeatureExtractor;
```

Comment:
>WordFeatureExtractor [](start = 47, length = 20)

Let's think about the "word feature extractors." This (and all similar things you have changed) were implementations of an interface, describing how to featurize text. It seemed like that simple solution was not only simpler than this sort of attempt to push this into a mutable state, but also significantly more powerful, and in a way that was hardly inappropriate given how many billions of ways have arisen over the years to tokenize text. If I wanted to make this easily configurable, I *might* have chosen to make that something that happened on the arguments object, rather than invent this separate concept -- it's strictly less powerful, a duplication of responsibilities, and also makes estimators *more* mutable, rather than less, which seems like going in the wrong direction.

Reply:
I didn't want to keep `Arguments` around, and this was my attempt at removing them.

I think I will just make the extractors hardcoded for now, and add flexibility back when we have the right estimators for them.

---
In reply to: [214522520](https://github.com/dotnet/machinelearning/pull/801#discussion_r214522520) [](ancestors = 214522520)